catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -211,34 +154,87 @@ public void setEnableBackupPlugin(boolean enablePlugin) {
      * @param dir absolute path for the root file system backup directory.
      */
     public void setRootBackupDir(String dir) {
-        if (StringUtils.isBlank(dir)) {
-            LOGGER.error("The root backup directory is blank.");
-            return;
+        Validate.notNull(dir);
+        File directory = new File(dir);
+        validateDirectory(directory);
+        this.rootBackupDir = directory;
+        LOGGER.trace("Set root backup directory to: {}", this.rootBackupDir.toString());
+    }
+
+    ExecutorService getExecutor() {
+        return executor;
+    }
+
+    void setExecutor(ExecutorService executor) {
+        this.executor = executor;
+    }
+
+    private synchronized void execute(Runnable task) {
```

Comment:
My understanding is that the `process(CreateResponse)`, `process(UpdateResponse)` and `process(DeleteResponse)` methods could be called by different threads, which means that order isn't guaranteed no matter what you do in the plugin. If we wanted to make sure those operations are done serially, the `CatalogFrameworkImpl` class would have to be changed. This is probably something you want to talk to @pklinef about.